core/services/functions/external_adapter_client.go

```diff
@@ -0,0 +1,94 @@
+package functions
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type ExternalAdapterInterface struct {
+	AdapterURL url.URL
+}
+
+type Payload struct {
+	RequestId            string `json:"requestId"`
+	JobName              string `json:"jobName"`
+	EncryptedSecretsUrls string `json:"encryptedSecretsUrls"`
+}
+
+type Response struct {
+	Result     string       `json:"result"`
+	Data       ResponseData `json:"data"`
+	StatusCode int          `json:"statusCode"`
+}
+
+type ResponseData struct {
+	Result      string `json:"result"`
+	Error       string `json:"error"`
+	ErrorString string `json:"errorString"`
+}
+
+func (ea ExternalAdapterInterface) FetchEncryptedSecrets(ctx context.Context, encryptedSecretsUrls []byte, requestId string, jobName string) (encryptedSecrets, userError []byte, err error) {
+	encodedSecretsUrls := base64.StdEncoding.EncodeToString(encryptedSecretsUrls)
+
+	payload := Payload{
+		RequestId:            requestId,
+		JobName:              jobName,
+		EncryptedSecretsUrls: encodedSecretsUrls,
+	}
+
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error constructing adapter encrypted secrets fetch payload")
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", ea.AdapterURL.JoinPath("/fetcher").String(), bytes.NewBuffer(jsonPayload))
```

Comment:
Can we require that the "/fetcher" path is just part of the URL and needs to be set by the caller?